llvm/lib/Transforms/Utils/SimplifyLibCalls.cpp

```diff
@@ -3109,6 +3109,32 @@ Value *LibCallSimplifier::optimizeRemquo(CallInst *CI, IRBuilderBase &B) {
   return ConstantFP::get(CI->getType(), Rem);
 }
 
+/// Constant folds fdim
+Value *LibCallSimplifier::optimizeFdim(CallInst *CI, IRBuilderBase &B) {
+  const APFloat *X, *Y;
+  // Check if both values are constants
+  if (!match(CI->getArgOperand(0), m_APFloat(X)) ||
+      !match(CI->getArgOperand(1), m_APFloat(Y)))
+    return nullptr;
```

Comment:
You don't do anything with the constant values, besides check if they are nan (which isn't strong enough to do this transform)

Reply:
not sure what you mean by that,
we need the NaN check and i don't see if we can do that through a `Value*`, we need to transform it to constant
can you suggest any alternatives here?

Reply:
You need to know a dynamic value cannot be nan, which isKnownNeverNaN can do